File: 二分/二分查找/744.py
class Solution(object):
    def nextGreatestLetter(self, letters, target):
        """
        :type letters: List[str]
        :type target: str
        :rtype: str
        """
        l, r = 0, len(letters) - 1
        while l <= r:
            mid = (l + r) // 2
            if ord(letters[mid]) >= ord(target) + 1:
                r = mid - 1
            else:
                l = mid + 1
        if l >= len(letters) or letters[l] <= target:
            return letters[0]
        else:
            return letters[l]
        

File: 每日一题/复健/744.cpp
#include <bits/stdc++.h>
using namespace std;
const int MOD = 1e9 + 7;

class Solution {
public:
    char nextGreatestLetter(vector<char>& letters, char target) {
        auto it = upper_bound(letters.begin(), letters.end(), target);
        if (it == letters.end()) return letters[0];
        return *it;
    }
};

// int main() {
//     vector<char> letters = {'c', 'f', 'j'};
//     char target;
//     cin >> target;
//     auto it = upper_bound(letters.begin(), letters.end(), target);
//     cout << (*it) << endl;
//     return 0;
// }